src/main/scala/inox/solvers/unrolling/ChooseEncoder.scala

```diff
@@ -25,6 +25,10 @@ trait ChooseEncoder extends transformers.ProgramTransformer {
 
     val newFds = sourceProgram.symbols.functions.values.toList.map { fd =>
       def rec(e: Expr, params: Seq[ValDef]): Expr = e match {
+        case l: Let =>
+          val free = exprOps.variablesOf(l)
+          l.copy(body = rec(l.body, params.filter(vd => free(vd.toVariable)) :+ l.vd)).copiedFrom(l)
```

Comment:
After a bit more thought, I think we should probably remove the filtering logic here (and in the `Lambda` case). I don't think it makes sense to restrict the variables chooses under lets/lambdas have access to.